File: binary-search/1800-concatenation-of-consecutive-binary-numbers/concatenation-of-consecutive-binary-numbers.py
class Solution:
    def concatenatedBinary(self, n: int) -> int:
        result = 0
        mod = 1_000_000_007;
        for i in range(1,n+1):
            length = i.bit_length()
            result = (result << length)%mod
            result = (result + i)%mod
        return result

File: binary-search/1800-concatenation-of-consecutive-binary-numbers/concatenation-of-consecutive-binary-numbers.java
class Solution {
    public int concatenatedBinary(int n) {
        long num = 0;
        int mod = 1_000_000_007;
        for(int i = 1;i<=n;i++){
            int length = Integer.toBinaryString(i).length();
            num = (num << length)%mod ;
            num = (num + i)%mod;
        }
        return (int)num;
    }
}